### ISSUE TITLE
Feature Question:

### ISSUE THREAD

**Issue by gn0rt0n** [plank/laravel-mediable#245]
Hi all, I love what you have done and this fits my needs very well as I need the many-to-many functionality that is provided. However, there are a couple of features from MediaLibrary that I like:
* Responsive Images (https://spatie.be/docs/laravel-medialibrary/v8/responsive-images/getting-started-with-responsive-images)
* Optimization (https://spatie.be/docs/laravel-medialibrary/v8/converting-images/optimizing-converted-images)

I did not see anything like this currently in the project. Have I missed them, and if not, are they on the roadmap by chance?

Thanks,
Gary

**Comment by frasmage** [comment]
Hi @gn0rt0n,

I'm glad that you are enjoying this package!

Regarding those two features:

**Responsive Images**: 
I had not considered doing something like that explicitly within the package, since the package is currently agnostic of the templating/presentation layer. That said, the "variants" feature of this package makes it easy to generate all of the necessary image sizes that you need for this, all that is missing is to map them to sizes in a img/srcset or picture element. For now, you could easily do something like this:

```
<img srcset="{{$media->findVariant('large')->getUrl()}} 600w, 
                        {{$media->findVariant('small')->getUrl()}} 300w"
         sizes="..."
         src="{{$media->findVariant('small')->getUrl()}}">
```

If you need more than this. I am happy to entertain PRs 🙂.

 **Optimization**: 
That's a great idea! I will need to investigate how to implement that, but it seems like a great feature addition.

**Comment by gn0rt0n** [comment]
@frasmage Thank you for the feedback. I appreciate the suggestion and clarification.

**Comment by frasmage** [comment]
Optimization is implemented in the upcoming major version release (#343)

**Comment by frasmage** [comment]
Version 6.0.0 has been released.
